### PR TITLE
Easier and better help for `io.ascii` `read` and `write` functions

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

This pull request is to make it easier to get help for `io.ascii.read/write()` and find format-specific options. The main idea is to follow the help precedent in the unified I/O -- add `io.ascii.read.help()` and `io.ascii.write.help()` functions just like `Table.read.help()` and `Table.write.help()`.

This also does some small re-ordering of the `read()` parameters in the documentation to put less-common args at the end.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/io/ascii/ui.py [hunk 1]
```diff
@@ -49,6 +49,67 @@
...
+        help_str,
+    )
+    help_str = re.sub(
+        r"Table\.(read|write)\(format='ascii'\)",
+        r"ascii.\1()",
+        help_str,
+    )
+
+    if out is None:
+        import pydoc
```
Comment (nstarman) [inline astropy/io/ascii/ui.py:77]: `pydoc` is a standard library. Any reason not to import at the top?

File: astropy/io/ascii/ui.py [hunk 2]
```diff
@@ -49,6 +49,67 @@
...
+    out : None or file-like
+        Output destination (default is stdout via a pager)
+    """
+
+
+def read_help(format=None, out=None):
+    _read_write_help("read", format=format, out=out)
+
+
+read_help.__doc__ = READ_WRITE_HELP.format(read_write="read")
```
Comment (nstarman) [inline astropy/io/ascii/ui.py:103]: We have https://docs.astropy.org/en/stable/api/astropy.utils.decorators.format_doc.html#astropy.utils.decorators.format_doc

File: astropy/io/ascii/ui.py [hunk 3]
```diff
@@ -49,6 +49,67 @@
...
+
+
+def write_help(format=None, out=None):
+    _read_write_help("write", format=format, out=out)
+
+
+write_help.__doc__ = READ_WRITE_HELP.format(read_write="write")
```
Comment (nstarman) [inline astropy/io/ascii/ui.py:110]: Maybe use https://docs.astropy.org/en/stable/api/astropy.utils.decorators.format_doc.html#astropy.utils.decorators.format_doc

**Review by nstarman — COMMENTED** [review]
A great addition! A few suggestions.

**Comment by hamogu** [comment]
I think this is a good idea. I'll let @taldcroft answer @nstarman before I approve.

**Comment by taldcroft** [comment]
@nstarman - thanks for the good comments, I think I've addressed them all. I didn't know about `format_doc`, there are other places I'd like to use that now.

**Review by hamogu — APPROVED** [review]

**Pull request merged**